files/initramfs-tools/union-mount.j2

```diff
@@ -85,5 +85,12 @@ then
   mount -t tmpfs -o rw,nosuid,nodev,size=${varlogsize}M tmpfs ${rootmnt}/var/log
   [ -f ${rootmnt}/host/disk-img/var-log.ext4 ] && rm -rf ${rootmnt}/host/disk-img/var-log.ext4
 else
+  [ -f ${rootmnt}/host/disk-img/var-log.ext4 ] && fsck.ext4 -v -p ${rootmnt}/host/disk-img/var-log.ext4 2>&1 \
+                                                  | gzip -c >> /tmp/fsck.log.gz
```

Comment:
>>> /tmp/fsck.log.gz [](start = 60, length = 19)

Why append instead of overwrite?

Reply:
If you only gzip 2 files, use 2 names and `mv` twice.

---
In reply to: [409184727](https://github.com/Azure/sonic-buildimage/pull/4431#discussion_r409184727) [](ancestors = 409184727)

Reply:
There are actually two ext3/ext4 filesystems that need to be checked:
1. the root filesystem mounted on the hard-drive or ssd, defined as root=X in /proc/cmdline
(this is done early from initrd by script fsck-rootfs)
2. the log filesystem, which is stored in an ext4 file [host/disk-img/var-log.ext4]
This is why we use one file, and we append to this file when the log fileystem is checked.